src/Display/Modal/Modal.tsx

```diff
@@ -26,34 +27,35 @@ class Modal extends React.Component<Props, State> {
     this.modalContentAreaRef = React.createRef();
   }
 
-  componentDidMount() {
-    const { onClose } = this.props;
-    document.addEventListener('keydown', escEvent(onClose), false);
-  }
+  componentDidUpdate(prevProps: Props) {
+    const { isVisible } = this.props;
 
-  static getDerivedStateFromProps(nextProps: Props, prevState: State) {
-    if (nextProps.isVisible && !prevState.isOpen) {
+    // closed -> open
+    if (!prevProps.isVisible && isVisible) {
+      this.modalContentAreaRef.current.focus();
```

Comment:
can we remove the `modalContentAreaRef`?

Reply:
I did not touch that logic in this PR. No idea what that's about in the first place haha.

Reply:
yeah, there seems to be some dead code and unnecessary code, like the `escEvent` which is essentially an alias of `onClose`, maybe can be removed as well.